sklearn/feature_extraction/tests/test_text.py

```diff
@@ -1042,6 +1045,27 @@ def test_vectorizer_string_object_as_input():
             ValueError, message, vec.transform, "hello world!")
 
 
+@pytest.mark.parametrize("X_dtype", [np.float32, np.float64])
+def test_tfidf_transformer_type(X_dtype):
+    X = rand(10, 20000, dtype=X_dtype, random_state=42)
+    X_trans = TfidfTransformer().fit_transform(X)
+    assert X_trans.dtype == X.dtype
+
+
+@pytest.mark.parametrize(
+    "vectorizer_dtype, output_dtype",
+    [(np.int32, np.float64),
```

Comment:
What's the sense in allowing an integer dtype to tfidf?

Reply:
This is the `dtype` of the vectorizer and not of the transformer. So it can be integer even if the transformer will return float.

Reply:
Yes, so essentially when `dtype` is not either float32 or float64 this parameter has no effect. Shouldn't a warning or error be raised in that case then (rather than silently ignoring a user-provided parameter)?

Reply:
The funny part here is that the default value of `dtype` is `np.int64`. It seems wrong then.
Shall we make it `np.float64`